spec.bs

```diff
@@ -386,49 +396,95 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
 <div algorithm>
   To <dfn>canonicalize protocol</dfn> given a string |value| and a string |type|:
 
-  1. TODO |value| |type|
+  1. Let |strippedValue| be the given |value| with a single trailing U+003A (:) removed, if any.
+  1. If |type| is "`pattern`" then return |strippedValue|.
+  1. Let |dummyURL| be a new [=url record=].
+  1. Let |parseResult| be the result of running the [=basic URL parser=] given |strippedValue| followed by "`://dummy.test`", with |dummyURL| as [=basic URL parser/url=].
```

Comment:
```suggestion
  1. Let |parseResult| be the result of running the [=basic URL parser=] given |strippedValue| followed by "`://dummy.test`", with |dummyURL| as <i>[=basic URL parser/url=]</i>.
```

Also for `[=basic URL parser/state override=]`. See https://infra.spec.whatwg.org/#algorithm-params and some background discussion in https://github.com/whatwg/infra/pull/325#issuecomment-702373931